### ISSUE TITLE
Trigger callback on error / warning 

### ISSUE THREAD

**Issue by vcnainala** [cheminfo/nmrium#1522]
Similar to the `onDataChange` event can we have an `onError` event emitted/raised (to trigger the parent component call back function)? 
Use case: This would be helpful to take subsequent actions when a spectral file doesn't load or any other warnings/errors.

**Comment by hamed-musallam** [comment]
@targos 

we need to bubble errors outside NMRium that we catch at a certain level of the tree nodes without being captured by   ErrorBoundary, for example, when we load a file not supported by NMRium we show a message to the user, currently, we need besides that to trigger callback function onError to be handled by an external component, I tried to implement a custom ErrorBoundary but this will not be a good solution because it re-rendering the children, do you think if we implement a context for this kind of errors is it going to be a good solution or what you suggest?

**Comment by targos** [comment]
I don't think we should rely on the ErrorBoundary for this. ErrorBoundary is for catching programming errors in nmrium (bugs), not for "external" user errors.

**Comment by targos** [comment]
Can you give me a way to reproduce an error like this?

**Comment by hamed-musallam** [comment]
@targos 

you can drag any file other than what nmrium support, for testing purpose .pdf will be sufficient

I rise an event for you https://github.com/cheminfo/nmrium/commit/89c294f4c256dcc57bb298e06e9369894f04e1bf

**Comment by lpatiny** [comment]
Anyway currently there is no error so a callback would have no effect and this should not throw any error. All the user errors should be managed in NMRium directly and NMRium should display a meaningful message to the user.

**Comment by hamed-musallam** [comment]
@lpatiny 

but maybe they need to take certain actions when the import failed even if the reason the file is not supported by NMRium?

**Comment by CS76** [comment]
> Anyway currently there is no error so a callback would have no effect and this should not throw any error. All the user errors should be managed in NMRium directly and NMRium should display a meaningful message to the user.

So, why isn't the spectra loading?? corrupt .jdx file? or isn't there any spectral data within the file?? In any case, as a user, I was expecting the spectra to be loaded and nothing happens which should technically be raising an error/warning about the missing data?

**Comment by lpatiny** [comment]
@CS76 Just open the file with a text editor. It is a mass spectrum

![image](https://user-images.githubusercontent.com/1484241/167074304-93c1b766-507f-4b1c-900d-c7946fcef93f.png)

**Comment by targos** [comment]
> but maybe they need to take certain actions when the import failed even if the reason the file is not supported by NMRium?

I'm not against supporting a callback to signal user errors or other kinds of messages to the calling component. I just think we should not rely on ErrorBoundary to do it. It can just be called from the code that handles spectra loading.
I suggest this: if the callback exists, call it. If not, the default callback shows an alert inside the component.

```jsx
<NMRium
  onEvent={(event) => console.log(event)}
/>
```

```ts
interface NMRiumEvent {
  kind: 'error'; // We could add other types in the future
  message: string;
}
```
